src/puppetlabs/puppetdb/import.clj

```diff
@@ -36,17 +60,32 @@
           ;;   the list of nodes that we submitted and the output of that query
           (command-fn :replace-catalog
                       (:replace_catalog command-versions)
-                      (utils/read-json-content tar-reader)))
+                      (extract-certname "catalogs" path)
+                      (-> tar-reader
+                          utils/read-json-content
+                          json/generate-string
+                          (.getBytes "UTF-8")
+                          java.io.ByteArrayInputStream.)))
       (file-pattern "reports")
       (do (log/infof "Importing report from archive entry '%s'" path)
           (command-fn :store-report
                       (:store_report command-versions)
-                      (utils/read-json-content tar-reader true)))
+                      (extract-certname "reports" path)
+                      (-> tar-reader
+                          (utils/read-json-content true)
```

Comment:
Not sure, I'd like to keep the keys as string in general, but in this case I just left alone what was there.